test/fork/merkle-lockup/MerkleLockupLD.t.sol

```diff
@@ -0,0 +1,232 @@
+    struct LeafData {
+        uint256 index;
+        uint256 recipientSeed;
+        LockupDynamic.SegmentWithDuration[] segments;
+    }
+
+    struct Params {
+        address admin;
+        uint40 expiration;
+        LeafData[] leafData;
+        uint256 posBeforeSort;
+    }
+
+    struct Vars {
+        uint256 actualStreamId;
+        LockupDynamic.Stream actualStream;
+        uint128[] amounts;
+        uint256 aggregateAmount;
+        uint128 clawbackAmount;
+        address expectedLockupLD;
+        MerkleLockup.ConstructorParams baseParams;
+        LockupDynamic.Stream expectedStream;
+        uint256 expectedStreamId;
+        uint256[] indexes;
+        uint256 leafPos;
+        uint256 leafToClaim;
+        ISablierV2MerkleLockupLD merkleLockupLD;
+        bytes32 merkleRoot;
+        address[] recipients;
+        uint256 recipientsCount;
+        uint40[] streamDurations;
+        LockupDynamic.SegmentWithDuration[][] segments;
+        LockupDynamic.Segment[][] segmentsWithTimestamps;
+    }
+
+    // We need the leaves as a storage variable so that we can use OpenZeppelin's {Arrays.findUpperBound}.
+    uint256[] public leaves;
+
+    function testForkFuzz_MerkleLockupLD(Params memory params) external {
+        vm.assume(params.admin != address(0) && params.admin != users.admin);
+        vm.assume(params.expiration == 0 || params.expiration > block.timestamp);
+        vm.assume(params.leafData.length > 1);
+        params.posBeforeSort = _bound(params.posBeforeSort, 0, params.leafData.length - 1);
+        assumeNoBlacklisted({ token: address(asset), addr: params.admin });
+
+        /*//////////////////////////////////////////////////////////////////////////
+                                          CREATE
+        //////////////////////////////////////////////////////////////////////////*/
+
+        Vars memory vars;
+
+        vars.recipientsCount = params.leafData.length;
+        vars.amounts = new uint128[](vars.recipientsCount);
+        vars.indexes = new uint256[](vars.recipientsCount);
+        vars.recipients = new address[](vars.recipientsCount);
+        vars.segments = new LockupDynamic.SegmentWithDuration[][](vars.recipientsCount);
+        vars.segmentsWithTimestamps = new LockupDynamic.Segment[][](vars.recipientsCount);
+        vars.streamDurations = new uint40[](vars.recipientsCount);
+
+        for (uint256 i = 0; i < vars.recipientsCount; ++i) {
+            vm.assume(
+                params.leafData[i].segments.length > 0
+                    && params.leafData[i].segments.length < defaults.MAX_SEGMENT_COUNT()
+            );
+            vars.indexes[i] = params.leafData[i].index;
+
+            // Bound each leaf's segment duration to avoid overflows.
+            fuzzSegmentDurations(params.leafData[i].segments);
+
+            uint256 segmentCount = params.leafData[i].segments.length;
+            for (uint256 j = 0; j < segmentCount; ++j) {
```

Comment:
Agree and updated.

Reply:
OK, fair enough if the reason is the readability of the test code. However, I want to mention that in terms of efficiencies, I think nested code could be more efficient than having two loops which can be more efficient than using `fuzzDynamicStreamAmounts` (3 loops). For the current size of the codebase, it does not matter. But when the codebase becomes much larger, this would speed up the fork tests.

Reply:
I have decided to use `fuzzDynamicStreamAmounts` for the code readability.